q2cli/core/config.py

```diff
@@ -0,0 +1,99 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2019, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import configparser
+
+import click
+
+import q2cli.util
+
+
+class CLIConfig():
+    path = os.path.join(q2cli.util.get_app_dir(), 'cli-colors.theme')
+    VALID_SELECTORS = {'options', 'type', 'default_arg', 'command', 'emphasis',
+                       'problem', 'errors', 'required', 'success'}
+    VALID_STYLINGS = {'fg', 'bg', 'bold', 'dim', 'underline', 'blink',
+                      'reverse'}
+    VALID_COLORS = {'black', 'red', 'green', 'yellow', 'blue', 'magenta',
+                    'cyan', 'white', 'bright_black', 'bright_red',
+                    'bright_green', 'bright_yellow', 'bright_blue',
+                    'bright_magenta', 'bright_cyan', 'bright_white'}
+    VALID_BOOLEANS = {'true': True,
+                      'false': False,
+                      't': True,
+                      'f': False}
+
+    def __init__(self):
+        self.styles = self._get_default_styles()
+        try:
+            self.parse_file(self.path)
+        except Exception:
+            if os.path.exists(self.path):
+                os.unlink(self.path)
+            self.styles = self._get_default_styles()
+
+    def _get_default_styles(self):
+        return {'options': {'fg': 'blue'},
+                'type': {'fg': 'green'},
+                'default_args': {'fg': 'magenta'},
```

Comment:
I am unable to roundtrip the generate-config and install-theme as the default style uses `default_args` but the VALID_SELECTORS wants `default_arg`.

Reply:
What about default GWAR?